src/test/scala/petclinic/services/AppointmentServiceSpec.scala

```diff
@@ -0,0 +1,106 @@
+package petclinic.services
+
+import io.github.scottweaver.zio.aspect.DbMigrationAspect
+import io.github.scottweaver.zio.testcontainers.postgres.ZPostgreSQLContainer
+import petclinic.models.Species
+import zio.test._
+
+import java.time.{LocalDate, LocalDateTime}
+
+object AppointmentServiceSpec extends DefaultRunnableSpec {
+
+  override def spec: Spec[TestEnvironment, TestFailure[Throwable], TestSuccess] = {
+    suite("AppointmentService")(
+      suite("added appointments exist in db")(
+        test("returns true confirming existence of added appointment") {
+          for {
+            owner <- OwnerService.create("Emily", "Elizabeth", "1 Birdwell Island, New York, NY", "212-215-1928")
+            pet <-
+              PetService.create("Clifford", LocalDate.of(1962, 2, 14), Species.fromString("Canine"), owner.id)
+            appointment <- AppointmentService.create(
+                             pet.id,
+                             LocalDateTime.of(2022, 6, 12, 13, 0),
+                             "Get updated body measurements"
+                           )
+            getAppointment <- AppointmentService.get(appointment.id)
+          } yield assertTrue(getAppointment.get == appointment)
+        },
+        test("returns true confirming existence of many added appointments") {
+          for {
+            owner <- OwnerService.create("Jon", "Arbuckle", "711 Maple St, Muncie, IN", "812-728-1945")
+            pet <-
+              PetService.create("Garfield", LocalDate.of(1978, 6, 19), Species.fromString("Feline"), owner.id)
+            appointment1 <-
+              AppointmentService.create(pet.id, LocalDateTime.of(2022, 7, 1, 9, 0), "Lasagna allergy test")
+            appointment2 <-
+              AppointmentService.create(pet.id, LocalDateTime.of(2022, 7, 11, 10, 0), "Monday allergy test")
+            appointments <- AppointmentService.getAll
+          } yield assertTrue(appointments.contains(appointment1) && appointments.contains(appointment2))
+        }
+      ),
+      suite("deleted appointments do not exist in db")(
+        test("returns true confirming non-existence of deleted appointment") {
+          for {
+            owner <-
+              OwnerService.create("Sherlock", "Holmes", "221B Baker St, London, England, UK", "+44-20-7224-3688")
+            pet <-
+              PetService.create("Toby", LocalDate.of(1888, 4, 17), Species.fromString("Canine"), owner.id)
```

Comment:
Good ol' Toby

Reply:
<img width="695" alt="CleanShot 2022-05-04 at 21 22 49@2x" src="https://user-images.githubusercontent.com/7587245/166851395-4c4a63d1-3f6a-4d74-9afd-e184f6ac368c.png">
Ah! You do your research 🕵️

Reply:
Tobylock Holmes